TrafficCapture/replayerPlugins/jsonMessageTransformers/Es2OsPost23/src/main/java/org/opensearch/migrations/transform/Es2OsPost23.java

```diff
@@ -1,9 +1,30 @@
 package org.opensearch.migrations.transform;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Es2OsPost23 implements IJsonTransformerProvider {
 
+    @Override
     public JsonJMESPathTransformer createTransformer(Object jsonConfig) {
-        String script = "{ \"settings\": settings, \"mappings\": payload.mappings.*.properties | [merge(@)] }";
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, Object> jsonMap = objectMapper.convertValue(jsonConfig, Map.class);
+        String script;
+
+        if (jsonMap.containsKey("mappings") && jsonMap.get("mappings") instanceof Map) {
+            Map<String, Object> mappings = (Map<String, Object>) jsonMap.get("mappings");
+            if (mappings.containsKey("properties")) {
+                // No type - no transformation needed for "mappings" since "properties" is directly under it.
+                script = "{\"settings\": settings, \"mappings\": mappings}";
```

Comment:
Once you cleanup the issues above, you can still have a Transformer that contains a couple JMESPath transformers within it as fields and switch which one you'd like to use within transformJson.